pkg/library/flatten/testdata/workspace-env/error_workspaceEnv-formatted-wrong.yaml

```diff
@@ -0,0 +1,29 @@
+name: "Returns an error when workspaceEnv cannot be read as []EnvVar"
+
+input:
+  devworkspace:
+    components:
+      - name: test-plugin
+        plugin:
+          uri: "https://my-plugin.io/test"
+  devfileResources:
+    "https://my-plugin.io/test":
+      schemaVersion: 2.0.0
+      metadata:
+        name: "plugin-a"
+      components:
+        - name: plugin-a
+          attributes:
+            # Note this fails to deserialize because it's just a yaml string, not a list.
+            # JSON can be used here if you remove the pipe
+            workspaceEnv: |
+              [{"name": "TEST_ENV_1",
+               "value": "TEST_VAL_1"},
+              {"name": "TEST_ENV_2",
+               "value": "TEST_VAL_2"}]
+          container:
+            name: test-container
+            image: test-image
+
+output:
+  errRegexp: "failed to read attribute workspaceEnv on component test-plugin"
```

Comment:
I know, it's not something that DWO controls but maybe it makes sense to include error as well, to understand better user-facing message. Currently it's going to be

```
failed to read attribute workspaceEnv on component test-plugin: json: cannot unmarshal string into Go value of type []v1alpha2.EnvVar
```

Reply:
Or even `failed to process workspaceEnv: failed to read attribute workspaceEnv on component test-plugin: json: cannot unmarshal string into Go value of type []v1alpha2.EnvVar
`

Reply:
I'm not strongly against testing the wording of errors, but I would want a better way to check. While the chance is small, this could ostensibly depending on Go version if we're looking for exact wording.

I'm also of the opinion that testing exact error messages is basically just writing the error message twice, but that's a more philosophical topic :)